### DISCUSSION TITLE
Start the fancy menu from a key binding from a wayland compositor

### DISCUSSION THREAD

**Discussion by romildo** [lxqt/lxqt#2658] [General]
Is it possible to start the fancy menu from a key binding in a wayland compositor?

I would like to have Alt+F1 bound to the LXQt fancy menu in LXQt+labwc, but I do not know which command would launch the fancy menu, so that it can be used to set the key binding.

**Comment by stefonarch** [comment]
There are only some hacks, one is using
```
qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
```
as command, but afaik in all compositors except Hyprland the menu has no keyboard focus, you can try it out.

Another hack is using `wlrctl` to move the cursor, here to top left:
```
wlrctl pointer move -2000 -1100 && sleep 0.2 && wlrctl pointer click
```

**Reply by romildo** [reply]
The first hack does not work for me on ArchLinux and NixOS:

```console
$ qdbus org.kde.StatusNotifierWatcher /global_key_shortcuts/panel/fancymenu/show_hide org.lxqt.global_key_shortcuts.client.activated
Error: org.freedesktop.DBus.Error.UnknownObject
No such object path '/global_key_shortcuts/panel/fancymenu/show_hide'
```

**Reply by tsujan** [reply]
> but afaik in all compositors except Hyprland the menu has no keyboard focus, you can try it out.

It has keyboard focus also on kwin_wayland. I think it should with Wayfire too, but maybe not with labwc.

> The first hack does not work for me on ArchLinux and NixOS:

A missing LXQt component?

**Reply by stefonarch** [reply]
No idea, on every arch system (I guess also on my debian) the path is present:
![immagine](https://github.com/user-attachments/assets/1ebf3427-ab99-43cc-9a06-7bf4729dca07)

Confirming: it works fine too on kwin and wayfire ; not on labwc, niri and sway.

**Reply by tsujan** [reply]
> Confirming: it works fine too on kwin and wayfire ; not on labwc, niri and sway.

kwin_wayland, hyprland and wayfire allowed us to make the panel unfocusable under them (with Qt ≥ 6.8). That should be the reason.

**Reply by romildo** [reply]
Somehow it is called `fancymenu2` instead of `fancymenu` on both ArchLinux and NixOS:

![image](https://github.com/user-attachments/assets/25722298-d4ad-4396-824a-137af0b114a0)

And it works.

**Reply by tsujan** [reply]
Oh, that's because of your `~/.config/lxqt/panel.conf`. Good catch.

**Reply by romildo** [reply]
Can I rename all occurrences of `fancymenu2` to `fancymenu` in `~/.config/lxqt/panel.conf` when not running LXQt to fix it?

**Reply by tsujan** [reply]
Yes, you could also do it when running LXQt, provided that you first stop lxqt-panel from LXQt Session Settings.

But it won't make a difference to the final result.

**Reply by stefonarch** [reply]
That happens if you create a second panel width fancymenu or add a second fancymenu.
